core/integration/gpu_test.go

```diff
@@ -0,0 +1,77 @@
+package core
+
+import (
+	"context"
+	"os/exec"
+	"regexp"
+	"strings"
+	"testing"
+
+	"dagger.io/dagger"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	cudaImage = "nvidia/cuda:11.7.1-base-ubuntu20.04"
+)
+
+var (
+	// uuidRegex is used to match GPU UUID in nvidia-smi output:
+	uuidRegex = regexp.MustCompile(`GPU-\w{8}-\w{4}-\w{4}-\w{4}-\w{12}`)
+)
+
+func TestGPUAccess(t *testing.T) {
+	ctx := context.Background()
+	c, err := dagger.Connect(ctx)
+	require.NoError(t, err)
+	defer c.Close()
+
+	// Query nvidia-smi on the host:
+	hostNvidiaCmd := exec.Command("nvidia-smi", "-L")
+	hostNvidiaOutput, err := hostNvidiaCmd.Output()
+	hostNvidiaOutputStr := string(hostNvidiaOutput)
+	require.NoError(t, err)
+	require.NotEmpty(t, hostNvidiaOutput)
+
+	// Query the same on the Dagger container and compare output:
+	ctr := c.Container().From(cudaImage)
+	contents, err := ctr.
+		// WithGPU(dagger.ContainerWithGPUOpts{Devices: "GPU-5d8950fe-17a6-2fa7-9baa-afa83bba0e2b"}).
+		WithGPU(dagger.ContainerWithGPUOpts{Devices: "all"}).
```

Comment:
I think I'd personally prefer if `all` wasn't a special string and instead we had a separate api call like `WithAllGPUs` (or similar, could instead be a bool option to `WithGPU` perhaps, though I like that less).

Just to cut back on the need for users to remember and type one-off strings like that correctly.